File: 🟡 Medium/3163 - String Compression III/_3163StringCompressionIII.js
/**
 * @param {string} word
 * @return {string}
 */
var compressedString = function (word) {
    let res = "";
    let i = 0;
    let curr = word[0];

    while (i < word.length) {
        let count = 0;

        while (i < word.length && word[i] == curr && count < 9) {
            i++;
            count++;
        }

        res += count + curr;

        if (i != word.length) {
            curr = word[i];
        }
    }

    return res;
};

File: 🟡 Medium/3163 - String Compression III/_3163StringCompressionIII.java
class Solution {
    public String compressedString(String word) {
        StringBuilder res = new StringBuilder("");
        int i = 0;
        char curr = word.charAt(0);

        while (i < word.length()) {
            int count = 0;

            while (i < word.length() && word.charAt(i) == curr && count < 9) {
                i++;
                count++;
            }

            res.append(count).append(curr);

            if (i != word.length()) {
                curr = word.charAt(i);
            }
        }

        return res.toString();
    }
}